packages/jellyfish-api-core/__tests__/category/loan/createLoanScheme.test.ts

```diff
@@ -0,0 +1,119 @@
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme default with same interestrate and mincolratio already exists\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if same id was created before', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'default' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme already exist with id default\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is an empty string', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: '' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is more than 8 chars long', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: '123456789' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
+  it('should createLoanScheme with utxos', async () => {
+    const masternodes = await testing.container.call('listmasternodes')
+    let masternodeId = ''
+    for (const id in masternodes) {
+      const masternode = masternodes[id]
+      if (masternode.mintedBlocks > 0) { // Find masternode that mined at least one block
+        masternodeId = id
+        break
+      }
+    }
+
+    const utxos = await testing.container.call('listunspent', [1, 9999999, [masternodes[masternodeId].ownerAuthAddress]])
+    const inputs: UTXO[] = utxos.map((utxo: UTXO) => {
+      return {
+        txid: utxo.txid,
+        vout: utxo.vout
+      }
+    })
+
+    const loanSchemeId = await testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'scheme1', utxos: inputs })
+    expect(typeof loanSchemeId).toStrictEqual('string')
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const rawtx = await testing.container.call('getrawtransaction', [loanSchemeId, true])
+    expect(rawtx.vin[0].txid).toStrictEqual(utxos[0].txid)
+    expect(rawtx.vin[0].vout).toStrictEqual(utxos[0].vout)
+
+    const data = await testing.container.call('listloanschemes')
+    const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+    expect(result.length).toStrictEqual(1)
+    expect(result[0]).toStrictEqual(
+      { id: 'scheme1', mincolratio: 200, interestrate: 2.5, default: false }
+    )
+  })
+
+  it('should not createLoanScheme with arbitrary utxos', async () => {
```

Comment:
```suggestion
  it('should not createLoanScheme with utxos not from foundation member', async () => {
```

Reply:
Will remember this. There is some code in Oracle side need to be refactored. Will do it together

Reply:
Fixed this.